Sources/Holocron/FileSystemStorageProvider.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  FileSystemStorageProvider.swift
+//  Holocron-iOS
+//
+//  Created by Pranjal Satija on 4/15/19.
+//  Copyright © 2019 CocoaPods. All rights reserved.
+//
+
+import Foundation
+
+// MARK: Base Struct
+public extension StorageProvider {
+    typealias FileSystem = FileSystemStorageProvider
+}
+
+public struct FileSystemStorageProvider {
+    let baseURL: URL
+    let readTransformer: Transformer
+    let writeTransformer: Transformer
+    
+    init(baseURL: URL, readTransformer: @escaping Transformer = { $0 }, writeTransformer: @escaping Transformer = { $0 }) {
+        self.baseURL = baseURL
+        self.readTransformer = readTransformer
+        self.writeTransformer = writeTransformer
+    }
+    
+    func url(for key: String) -> URL {
+        return baseURL.appendingPathComponent(key)
+    }
+    
+    typealias Transformer = (Data) -> Data
+}
+
+// MARK: Public API
+extension FileSystemStorageProvider {
+    /// Writes a value for a specific key.
+    /// - parameter value: The value to store.
+    /// - parameter key: The key to associate with `value`.
+    /// - parameter options: The options to use when writing the file.
+    public func write<T: Encodable>(_ value: T, for key: String, options: Data.WritingOptions) throws {
```

Comment:
It might also be cool to be able to customize `FileProtectionType` as well.

Reply:
So instead of passing in `options`, we ask for a `FileProtectionType` directly?

Reply:
Another symptom of today - `Data.WritingOptions` covers `FileProtectionType` and atomicity. Nevermind 👍